pkg/cmd/info.go

```diff
@@ -22,16 +32,27 @@ var (
 )
 
 type InfoOptions struct {
+	printFlags  *genericclioptions.PrintFlags
 	configFlags *genericclioptions.ConfigFlags
 
+	discoveryClient discovery.CachedDiscoveryInterface
+	dynamicClient   dynamic.Interface
+
+	printer printers.ResourcePrinter
 	builder *resource.Builder
 	args    []string
 
+	// directory where all gathered data will be stored
+	baseDir string
+	// whether or not to allow writes to an existing and populated base directory
+	overwrite bool
+
 	genericclioptions.IOStreams
 }
 
 func NewInfoOptions(streams genericclioptions.IOStreams) *InfoOptions {
 	return &InfoOptions{
+		printFlags:  genericclioptions.NewPrintFlags("gathered").WithDefaultOutput("yaml"),
```

Comment:
@deads2k I should probably set a `WithTypeSetter()` param here, right? This might save me the trouble of manually calling `SetGroupVersionKind()` on each resource I want to print later. Not sure what scheme to use, however.

Reply:
> @deads2k I should probably set a `WithTypeSetter()` param here, right? This might save me the trouble of manually calling `SetGroupVersionKind()` on each resource I want to print later. Not sure what scheme to use, however.

a scheme that matches the client you plan to use.